runtime/compiler/runtime/RelocationRuntime.cpp

```diff
@@ -1404,3 +1404,44 @@ TR_JITServerRelocationRuntime::copyDataToCodeCache(const void *startAddress, siz
    return coldCodeStart;
    }
 #endif /* defined(J9VM_OPT_JITSERVER) */
+
+void
+printAOTHeader(const void* aotHeaderAddress, char * buff, const size_t BUFF_SIZE)
+   {
+   if (!aotHeaderAddress)
+      {
+      strncat(buff, "null", buffSize);
+      return;
+      }
+
+   PORT_ACCESS_FROM_PORT(TR::Compiler->portLib);
+   OMRPORT_ACCESS_FROM_OMRPORT(TR::Compiler->omrPortLib);
+   TR_AOTHeader * hdrInCache = (TR_AOTHeader *)aotHeaderAddress;
+   OMRProcessorDesc processorDescription = hdrInCache->processorDescription;
+
+   int rowLength = 0;
+   for (size_t i = 0; i < OMRPORT_SYSINFO_FEATURES_SIZE; i++)
+      {
+      for (int j = 0; j < 32; j++) 
+         {
+         if (processorDescription.features[i] & (1<<j))
+            {
+            if (rowLength >= 20)
+               {
+               strncat(buff, "\n\t                                       ", BUFF_SIZE - strlen(buff) - 1);
```

Comment:
You should memset buff to 0 before entering the outermost for loop. Otherwise, depending on the stack, the first time you call `strlen(buff)` you might not get 0.